### PR TITLE
WIP: mplex implementation.

### PR DESCRIPTION
**Pull request by snazha-blkio**

### CODE REVIEW

**Comment by Nashatyrev** [comment]
I think it may have sense to merge `feature/multistream-handler` to your PR branch. 
I will delete obsolete `sketch` branch if nobody minds

**Comment by snazha-blkio** [comment]
> I think it may have sense to merge `feature/multistream-handler` to your PR branch.
> I will delete obsolete `sketch` branch if nobody minds

Thanks @Nashatyrev , I've merged the `feature/multistream-handler` branch into this branch.

**Comment by Nashatyrev** [comment]
@snazha-blkio Thanks for PR, could you please resolve lint failures?

**Comment by Nashatyrev** [comment]
Can you please merge `feature/multistream-handler` again (I've just merged there a couple of fixes) and rebase this PR from `master` to  `feature/multistream-handler` ?

File: src/main/kotlin/io/libp2p/core/mplex/MultiplexStreamState.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 BLK Technologies Limited (web3labs.com).
```
Comment (conor10) [inline src/main/kotlin/io/libp2p/core/mplex/MultiplexStreamState.kt]: We're now Web3 Labs Ltd

**Pull request merged**